webapp/src/containers/createRequest/index.tsx

```diff
@@ -0,0 +1,197 @@
+
+import {resetGlobalModalState} from 'src/reducers/globalModal';
+import {isCreateRequestModalOpen} from 'src/selectors';
+
+import './styles.scss';
+
+// TODO: remove later after integration with the APIs
+const requestOptions: RequestData[] = [
+    {
+        sys_id: 'sys_id 1',
+        name: 'name 1',
+        short_description: 'desc 1',
+        price: 'price 1',
+        category: {
+            sys_id: 'category 1 sys_id',
+            title: 'category 1',
+        },
+    },
+    {
+        sys_id: 'sys_id 2',
+        name: 'name 2',
+        short_description: 'desc 2',
+        price: 'price 2',
+        category: {
+            sys_id: 'category 2 sys_id',
+            title: 'category 2',
+        },
+    },
+    {
+        sys_id: 'sys_id 3',
+        name: 'item 1',
+        short_description: 'desc 3',
+        price: 'price 3',
+        category: {
+            sys_id: 'category 3 sys_id',
+            title: '',
+        },
+    },
+    {
+        sys_id: 'sys_id 4',
+        name: 'itme 2',
+        short_description: 'desc 4',
+        price: 'price 4',
+        category: {
+            sys_id: 'category 4 sys_id',
+            title: '',
+        },
+    },
+];
+
+const CreateRequest = () => {
+    const [options, setOptions] = useState<RequestData[]>(requestOptions);
+    const [suggestions, setSuggestions] = useState<Record<string, string>[]>([]);
+    const [autoSuggestValue, setAutoSuggestValue] = useState('');
+    const [suggestionChosen, setSuggestionChosen] = useState(false);
+    const [request, setRequest] = useState<Record<string, string> | null>(null);
+
+    // Loaders
+    const [showModalLoader, setShowModalLoader] = useState(false);
+
+    // usePluginApi hook
+    const {getApiState, pluginState} = usePluginApi();
+
+    // Errors
+    const [apiError, setApiError] = useState<APIError | null>(null);
+
+    const dispatch = useDispatch();
+
+    const hideModal = useCallback(() => {
+        setOptions([]);
+        setSuggestions([]);
+        setAutoSuggestValue('');
+        setApiError(null);
+        setSuggestionChosen(false);
+        setRequest(null);
+        dispatch(resetGlobalModalState());
+    }, []);
+
+    const mapRequestsToSuggestions = (requests: RequestData[]): Array<Record<string, string>> => requests.map((r) => ({
+        id: r.sys_id,
+        name: r.name,
+        short_description: r.short_description,
+        price: r.price,
+        title: r.category.title,
+        category_id: r.category.sys_id,
+    }));
+
+    // Set the suggestions when the input value of the auto-suggest changes;
+    useEffect(() => {
+        const requestsToSuggest = options?.filter((r) => r.name.toLowerCase().includes(autoSuggestValue.toLowerCase())) || [];
+        setSuggestions(mapRequestsToSuggestions(requestsToSuggest));
+    }, [autoSuggestValue]);
+
+    useEffect(() => {
+        // When the request value is reset, reset the caller auto-suggest input as well;
```

Comment:
Is there need any of semi colons in the end of comments ?
```suggestion
        // Reset the caller auto-suggest input when the request value is reset.
```

Reply:
updated